packages/migration/src/mods/7.0.0/renameReactDeprecated.ts

```diff
@@ -0,0 +1,25 @@
+import { mod } from 'riceburn';
+import ts from 'typescript';
+import { migration } from '../../migration';
+
+export default migration('rename componentWillReceiveProps with UNSAFE_componentWillReceiveProps', () => {
+  mod('**/*.ts?(x)').asTypescript((node, modder) => {
+    const replacements = {
+      componentWillMount: 'UNSAFE_componentWillMount',
+      componentWillReceiveProps: 'UNSAFE_componentWillReceiveProps',
+      componentWillUpdate: 'UNSAFE_componentWillUpdate'
+    };
+
+    if (ts.isClassDeclaration(node)) {
+      for (const member of node.members) {
+        if (member.name) {
+          for (const from of Object.keys(replacements) as (keyof typeof replacements)[]) {
```

Comment:
I've been hitting this too lately...no idea why TS can't just infer the type here... :(

Reply:
Yes. This is a problem in the typings of object.keys... They didn't make this thing generic so that it can be inferred. I wonder if we can make a or for definitely typed.